### DISCUSSION TITLE
How to directly connect nodes without the edge creation tool from the tool palette?

### DISCUSSION THREAD

**Discussion by TheMicDiet** [eclipse-glsp/glsp#768] [Q&A] [answered]
Hey, 
I was wondering how to connect nodes directly without having to select the edge creation tool from the tool palette. I saw that there is an example in your gallery (https://www.eclipse.org/glsp/gallery/#portsandrouting) that looks like what I need. Is there any documentation on how to implement this?

**Comment by martin-fleck-at** [comment, marked as answer]
@TheMicDiet Unfortunately, we do not have that example online. However, the basic idea is that the tool manager in GLSP ensures which tools are enabled or disabled. Tools can be registered as default tool or "regular" tool whereas a default tool is enabled by default (e.g., the selection tool which registers a mouse listener to adapt the selected state). This behavior can be managed through the `EnableToolsAction` and the `EnableDefaultToolsAction`.

When you click on one of the icons in the tool palette, that particular tool gets enabled and all other tools are disabled (mainly to avoid conflicts between the tools). So to get the behavior you describe, you need to register a connection tool as default tool. That tool can then register a mouse listener on its enablement where you can listen to clicks, dragging, etc. to ensure you get the correct elements that you aim to connect.

This should get you started on the right track.
